cloud/aws/deploy/batch.go

```diff
@@ -15,13 +15,348 @@
 package deploy
 
 import (
+	"encoding/json"
 	"fmt"
+	"strconv"
 
+	"github.com/nitrictech/nitric/cloud/common/deploy/image"
 	"github.com/nitrictech/nitric/cloud/common/deploy/provider"
+	"github.com/nitrictech/nitric/cloud/common/deploy/tags"
 	deploymentspb "github.com/nitrictech/nitric/core/pkg/proto/deployments/v1"
+	"github.com/pulumi/pulumi-awsx/sdk/go/awsx/ec2"
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+
+	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/batch"
+	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/ecr"
+	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/iam"
+	awsec2 "github.com/pulumi/pulumi-aws/sdk/v6/go/aws/ec2"
 )
 
-func (p *NitricAwsPulumiProvider) Batch(ctx *pulumi.Context, parent pulumi.Resource, name string, config *deploymentspb.Batch, runtimeProvider provider.RuntimeProvider) error {
-	return fmt.Errorf("Not implemented")
+type ResourceRequirement struct {
+	Type  string `json:"type"`
+	Value string `json:"value"`
+}
+
+type EnvironmentVariable struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+func (a *NitricAwsPulumiProvider) batch(ctx *pulumi.Context) error {
+	var subnets pulumi.StringArrayOutput
+	var vpcId pulumi.StringOutput
+	var err error
+
+	if a.Vpc != nil {
+		allSubnets := allVpcSubnetIds(a.Vpc)
+
+		subnets = allSubnets
+		vpcId = a.Vpc.VpcId
+	} else {
+		vpc, err := ec2.NewDefaultVpc(ctx, "default-vpc", nil)
+		if err != nil {
+			return fmt.Errorf("could not resolve default VPC")
+		}
+
+		subnets = vpc.PublicSubnetIds
+		vpcId = vpc.VpcId
+	}
+
+	a.BatchSecurityGroup, err = awsec2.NewSecurityGroup(ctx, "batch-sg", &awsec2.SecurityGroupArgs{
+		VpcId: vpcId,
+		Egress: awsec2.SecurityGroupEgressArray{
+			awsec2.SecurityGroupEgressArgs{
+				Protocol: pulumi.String("-1"),
+				FromPort: pulumi.Int(0),
+				ToPort:   pulumi.Int(0),
+				// Still need public internet access for batch jobs by default
+				// TODO: Allow restriction via config
+				CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
```

Comment:
Can this be resolved?

Reply:
Possibly, I don't have a good solution for it, but it should be fine for preview.